src/gui/properties/trackerlist.h

```diff
@@ -31,60 +31,78 @@
 #ifndef TRACKERLIST_H
 #define TRACKERLIST_H
 
+#include <QClipboard>
+#include <QList>
 #include <QShortcut>
 #include <QTreeWidget>
-#include <QList>
-#include <QClipboard>
 
 #include "propertieswidget.h"
 
-enum TrackerListColumn {COL_TIER, COL_URL, COL_STATUS, COL_RECEIVED, COL_SEEDS, COL_PEERS, COL_DOWNLOADED, COL_MSG};
 #define NB_STICKY_ITEM 3
 
 namespace BitTorrent
 {
     class TorrentHandle;
 }
 
-class TrackerList: public QTreeWidget {
-  Q_OBJECT
-  Q_DISABLE_COPY(TrackerList)
-
-private:
-  PropertiesWidget *properties;
-  QHash<QString, QTreeWidgetItem*> tracker_items;
-  QTreeWidgetItem* dht_item;
-  QTreeWidgetItem* pex_item;
-  QTreeWidgetItem* lsd_item;
-  QShortcut *editHotkey;
-  QShortcut *deleteHotkey;
-  QShortcut *copyHotkey;
+class TrackerList: public QTreeWidget
+{
+    Q_OBJECT
+    Q_DISABLE_COPY(TrackerList)
 
 public:
-  TrackerList(PropertiesWidget *properties);
-  ~TrackerList();
+    enum TrackerListColumn
+    {
+        COL_TIER,
+        COL_URL,
+        COL_STATUS,
+        COL_RECEIVED,
+        COL_SEEDS,
+        COL_PEERS,
+        COL_DOWNLOADED,
+        COL_MSG,
 
-protected:
-  QList<QTreeWidgetItem*> getSelectedTrackerItems() const;
+        COL_COUNT
+    };
+
+    TrackerList(PropertiesWidget *properties);
+    ~TrackerList();
+
+    int visibleColumnsCount() const;
 
 public slots:
-  void setRowColor(int row, QColor color);
-
-  void moveSelectionUp();
-  void moveSelectionDown();
-
-  void clear();
-  void loadStickyItems(BitTorrent::TorrentHandle *const torrent);
-  void loadTrackers();
-  void askForTrackers();
-  void copyTrackerUrl();
-  void reannounceSelected();
-  void deleteSelectedTrackers();
-  void editSelectedTracker();
-  void showTrackerListMenu(QPoint);
-  void loadSettings();
-  void saveSettings() const;
+    void setRowColor(int row, QColor color);
+
+    void moveSelectionUp();
+    void moveSelectionDown();
+
+    void clear();
+    void loadStickyItems(BitTorrent::TorrentHandle *const torrent);
+    void loadTrackers();
+    void askForTrackers();
+    void copyTrackerUrl();
+    void reannounceSelected();
+    void deleteSelectedTrackers();
+    void editSelectedTracker();
+    void showTrackerListMenu(QPoint);
+    void displayToggleColumnsMenu(const QPoint &);
+    void loadSettings();
+    void saveSettings() const;
+
+protected:
+    QList<QTreeWidgetItem *> getSelectedTrackerItems() const;
+
+private:
+    PropertiesWidget *properties;
```

Comment:
One more requests. Can you fix coding style of these private names? Please change to camelCase and add `m_` prefix.